code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm

```diff
@@ -0,0 +1,241 @@
+/obj/structure/cult
+	density = 1
+	anchored = 1
+	icon = 'icons/obj/cult.dmi'
+	var/health = 50
+	var/maxHealth = 50
+	var/sound_damaged = null
+	var/sound_destroyed = null
+
+
+//if you want indestructible buildings, just make a custom takeDamage() proc
+/obj/structure/cult/proc/takeDamage(var/damage)
+	health -= damage
+	if (health <= 0)
+		if (sound_destroyed)
+			playsound(get_turf(src), sound_destroyed, 100, 1)
+		qdel(src)
+	else
+		update_icon()
+
+/obj/structure/cult/New()
+	..()
+	flick("[icon_state]-spawn", src)
+
+/obj/structure/cult/Destroy()
+	flick("[icon_state]-break", src)
+	..()
+
+//duh
+/obj/structure/cult/cultify()
+	return
+
+//nuh-uh
+/obj/structure/cult/acidable()
+	return 0
+
+/obj/structure/cult/ex_act(var/severity)
+	switch(severity)
+		if (1)
+			takeDamage(100)
+		if (2)
+			takeDamage(20)
+		if (3)
+			takeDamage(4)
+
+/obj/structure/cult/blob_act()
+	playsound(get_turf(src), sound_damaged, 75, 1)
+	takeDamage(20)
+
+/obj/structure/cult/bullet_act(var/obj/item/projectile/Proj)
+	takeDamage(Proj.damage)
+	..()
+
+/obj/structure/cult/attackby(var/obj/item/weapon/W, var/mob/user)
+	if (istype(W, /obj/item/weapon/grab))
+		if(iscarbon(W:affecting))
+			MouseDrop_T(W:affecting,user)
+			returnToPool(W)
+	else if (istype(W, /obj/item/weapon))
+		if(user.a_intent == I_HURT)
+			user.delayNextAttack(8)
+			if (sound_damaged)
+				playsound(get_turf(src), sound_damaged, 75, 1)
+			takeDamage(W.force)
+			..()
+		else
+			MouseDrop_T(W,user)
+
+
+/obj/structure/cult/attack_paw(var/mob/user)
+	return attack_hand(user)
+
+
+/obj/structure/cult/attack_hand(var/mob/living/user)
+	if(user.a_intent == I_HURT)
+		user.delayNextAttack(8)
+		user.visible_message("<span class='danger'>[user.name] kicks \the [src]!</span>", \
+							"<span class='danger'>You kick \the [src]!</span>", \
+							"You hear stone cracking.")
+		takeDamage(user.get_unarmed_damage(src))
+		if (sound_damaged)
+			playsound(get_turf(src), sound_damaged, 75, 1)
+	else if(iscultist(user))
+		cultist_act(user)
+	else
+		noncultist_act(user)
+
+/obj/structure/cult/proc/cultist_act(var/mob/user)
+	if(!iscultist(user))//just to be extra safe
+		return 0
+	return 1
+
+/obj/structure/cult/proc/noncultist_act(var/mob/user)
```

Comment:
Remember the cultist chaplain religion, could be fun for here

Reply:
Yeah I made a reference to it in a comment like 4 lines bellow. There'll be some polish once every Cult 3.0 feature is functional.